typespec-extension/src/code-model-builder.ts

```diff
@@ -1072,14 +1074,23 @@ export class CodeModelBuilder {
         schema = this.processSchemaFromSdkType(sdkType, param.param.name);
       }
 
-      // skip-url-encoding
       let extensions: { [id: string]: any } | undefined = undefined;
+      // skip-url-encoding
+      if (param.type === "path") {
+        const pathParamOptions = getPathParamOptions(this.program, param.param);
+        if (pathParamOptions.allowReserved) {
+          extensions = extensions ?? {};
+          extensions["x-ms-skip-url-encoding"] = true;
+        }
+      }
+      // TODO: deprecate this logic of string/url for x-ms-skip-url-encoding
       if (
         (param.type === "query" || param.type === "path") &&
         param.param.type.kind === "Scalar" &&
         schema instanceof UriSchema
       ) {
-        extensions = { "x-ms-skip-url-encoding": true };
+        extensions = extensions ?? {};
+        extensions["x-ms-skip-url-encoding"] = true;
       }
```

Comment:
I didn't deprecate the old logic of `uri for skip encoding, string for not skip encoding`.

Plan to check later.

Reply:
Yeah, TCGC should handle it and returns correct collection format. I am integrating with latest TCGC. I will pay attention to the test case.